test/ml_kem_internal_test.c

```diff
@@ -107,8 +107,10 @@ static int sanity_test(void)
         return 0;
 
     if (!TEST_ptr(privctx = RAND_get0_private(NULL))
-        || !TEST_ptr(pubctx = RAND_get0_public(NULL)))
+        || !TEST_ptr(pubctx = RAND_get0_public(NULL))) {
+        EVP_MD_free(sha256);
```

Comment:
There is another return in the loop also.

Reply:
It's probably worth adding an `err` labels down the bottom and using `goto` to go there.